priority_queue_2.py

```diff
@@ -0,0 +1,58 @@
+from binheap import BinHeap
+
+
+class Node:
+    def __init__(self, value=None, priority=None, insertion_order=None):
```

Comment:
I'm curious why the insertion_order is needed here?  A binary heap should be fifo already.  Is there something wrong with the heap?

Reply:
The heap is not fifo, it has no respect towards the order of insertion, only the priority(value), items are often moved around to maintain heap structure and order of insertion is not considered during this. Correct me if i'm wrong, but I didn't think a heap was supposed to follow any specific order other than priority(value).